app/jobs/create_rss_notifications.rb

```diff
@@ -1,6 +1,6 @@
 class CreateRssNotifications < ApplicationJob
   after_perform do |job|
-    self.class.set(:wait => 12.hours).perform_later(job.arguments.first)
+    self.class.set(:wait => 12.hours).perform_later(job.arguments)
```

Comment:
I added pry here and was surprised that `job.arguments` is actually an array, because ActiveJob::Base [stores it as an array](https://github.com/rails/rails/blob/master/activejob/lib/active_job/core.rb#L71-L72). Therefore the `job.arguments` is `[{}]` and therefore the `arguments.first` was required. When someone calls it e.g. from console without any param, empty array is passed so `job.arguments.first` would resolve to nil. So in fact this part needs to be a bit more smarter and unfold the arguments, so it should look like this `perform_later(*jobs.arguments)`. That way we ensure all arguments passed to previous `perform` are passed to `perform_later`.

Sorry for wrong suggestion at first, but we're getting there :-)

Reply:
i hope that now it is okay :D

Reply:
I'm afraid it's not, `job.arguments` is the array of arguments that were passed to perform, imagine you create a job like this `CreateRssNotifications(1, 2, 3)`, then the `job.arguments # => [1,2,3]` so the original code would perform the job later but only with 1 throwing away all the rest. Therefore the `job.argument.first` is wrong. Defaulting to empty hash would again be just a workaround for case no arguments were passed to the job perform method.

We should pass the very same set of arguments, therefore we need to unfold it using `*`. So `perform_later(*job.arguments)` is IMHO the only correct call. That handles the empty set of params too.

Reply:
right, i see what you mean, now b/c there is no value at all inside the array, the arity will be 0, and the default hash will take place.

Done :)

Reply:
exactly, I'll wait for tests, do quick tests and then it's good to go :-)